### PR TITLE
Make locale parsing and converting more robust. Fixes #3375

### LINKED ISSUE #3375: 500 error caused by locals at webextension submission
**Steps to reproduce:**
Try to submit the attached webextension:
[Any.do-Extension_v3.1.3.22.zip](https://github.com/mozilla/addons-server/files/447548/Any.do-Extension_v3.1.3.22.zip)

**Expected results:**
The webextension is successfully submitted.

**Actual results:**
Server error is returned and the oops message is displayed.

**Additional notes:**
- This issue reproduces on AMO-dev, AMO-stage and AMO-production using Firefox 51.0a1 (2016-08-31) under Windows 10 64-bit.
- I suspect that this issue is caused by the “x” character from locales folders name because the error is no longer displayed if I removed the “x_”. 
- Please see the attached video:
gif

### PR DESCRIPTION
**Pull request by EnTeQuAk**
Fixes #3375

### CODE REVIEW

File: src/olympia/amo/tests/test_amo_utils.py [hunk 1]
```diff
@@ -97,7 +97,8 @@ def test_resize_transparency_for_P_mode_bug_1181221():
     ('en_US', 'en-US'),
     ('en_us', 'en-US'),
     ('FR', 'fr'),
-    ('el', 'el')
+    ('el', 'el'),
+    ('x_zh_cn', 'x-ZH'),  # see mozilla/addons-server#3375
```
Comment (tofumatt) [inline src/olympia/amo/tests/test_amo_utils.py]: Could you make these links to the issues instead? It's possible someone might not know what this means... but even if they did it's easy to have something I can click/copy+paste.

**Comment by tofumatt** [comment]
I didn't know locales could look like that! r+

**Comment by EnTeQuAk** [comment]
> I didn't know locales could look like that! r+

Well locales can contain all sorts of stuff - this here in particular is wrong though but we failed with a 500 before and we shouldn't but more or less just don't accept the data (which is what we do here, convert it to something invalid so that we ignore it everywhere, e.g in amo.utils:find_language)

**Comment by EnTeQuAk** [comment]
This will probably need a few more tests for locales like `sr_Cyrl_BA` (Serbian (Cyrillic, Bosnia and Herzegovina)) or `zh_Hans_CN`(Chinese (Simplified Han, China)) to make sure we behave correctly.

**Pull request merged**